packages/svelte/src/compiler/phases/2-analyze/visitors/RenderTag.js

```diff
@@ -13,13 +13,31 @@ export function RenderTag(node, context) {
 	validate_opening_tag(node, context.state, '@');
 
 	node.metadata.path = [...context.path];
-	context.state.analysis.elements.push(node);
 
 	const callee = unwrap_optional(node.expression).callee;
 
-	node.metadata.dynamic =
-		callee.type !== 'Identifier' || context.state.scope.get(callee.name)?.kind !== 'normal';
+	const binding = callee.type === 'Identifier' ? context.state.scope.get(callee.name) : undefined;
+
+	node.metadata.dynamic = binding?.kind !== 'normal';
+
+	/** If we can't unambiguously resolve this to a declaration, we
+	 * must assume the worst and link the render tag to every snippet
+	 */
+	let resolved =
+		callee.type === 'Identifier' &&
+		(!binding ||
+			binding.declaration_kind === 'import' ||
+			binding.kind === 'prop' ||
+			binding.kind === 'rest_prop' ||
+			binding.kind === 'bindable_prop' ||
+			binding?.initial?.type === 'SnippetBlock');
```

Comment:
this is almost the same code as in `component.js` , feels like that can be extracted into a helper function (along with a comment explaining why exactly this chain is done)